server/src/main/java/org/candlepin/async/tasks/EntitleByProductsJob.java

```diff
@@ -0,0 +1,183 @@
+import org.candlepin.controller.Entitler;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Owner;
+
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Asynchronous job for entitlement by products. A job will wait for a running
+ * job of the same Owner to finish before beginning execution
+ */
+public class EntitleByProductsJob implements AsyncJob {
+
+    private static Logger log = LoggerFactory.getLogger(EntitleByProductsJob.class);
+    protected Entitler entitler;
+    protected ConsumerCurator consumerCurator;
+
+    public static final String JOB_KEY = "ENTITLE_BY_PRODUCTS";
+    private static final String JOB_NAME = "bind_by_products";
+
+    private static final String OWNER_KEY = "org";
+    private static final String CONSUMER_UUID_KEY = "consumer_uuid";
+    private static final String ENTITLE_DATE_KEY = "entitle_date";
+    private static final String PROD_IDS_KEY = "product_ids";
+    private static final String FROM_POOLS_KEY = "from_pools";
+
+    @Inject
+    public EntitleByProductsJob(final Entitler e, final ConsumerCurator c) {
+        entitler = e;
+        consumerCurator = c;
+    }
+
+    @Override
+    public Object execute(final JobExecutionContext ctx) throws JobExecutionException {
+        try {
+            final JobArguments arguments = ctx.getJobArguments();
+            final String consumerUuid = arguments.getAsString(CONSUMER_UUID_KEY);
+            final Date entitleDate = arguments.getAs(ENTITLE_DATE_KEY, Date.class);
+            final String[] prodIds = arguments.getAs(PROD_IDS_KEY, String[].class);
+            final String[] pools = arguments.getAs(FROM_POOLS_KEY, String[].class);
+
+            final List<Entitlement> ents = entitler.bindByProducts(
+                prodIds, consumerUuid, entitleDate, Arrays.asList(pools));
+            entitler.sendEvents(ents);
+            return "Entitlements created for owner";
+        }
+        // Catch any exception that is fired and re-throw as a JobExecutionException
+        // so that the job will be properly cleaned up on failure.
+        catch (Exception e) {
+            log.error("EntitlerJob encountered a problem.", e);
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    public static EntitleByProductsJobConfig createConfig() {
+        return new EntitleByProductsJobConfig();
+    }
+
+    /**
+     * Job configuration object for the entitle by products job
+     */
+    public static class EntitleByProductsJobConfig extends JobConfig {
+
+        public EntitleByProductsJobConfig() {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME)
+                .addConstraint(JobConstraints.uniqueByArguments(OWNER_KEY));
```

Comment:
The original job does not seem to extend `UniqueByEntityJob`, or implement the `isSchedulable/scheduleJob` methods, so there is no need for constraints on this job.

Reply:
Done.